test/parallel/test-http2-client-rststream-before-connect.js

```diff
@@ -15,12 +15,14 @@ server.on('listening', common.mustCall(() => {
   const client = h2.connect(`http://localhost:${server.address().port}`);
 
   const req = client.request({ ':path': '/' });
+  // make sure that destroy is called twice
+  req._destroy = common.mustCall(req._destroy.bind(req), 2);
+
+  client.rstStream(req, 0);
+  // redundant call to rstStream to test rst won't be called again
   client.rstStream(req, 0);
```

Comment:
If you call the second `client.rstStream` with a different code, you could then verify that the `rstCode` didn't get set to the 2nd incorrect code. Right now it's a bit ambiguous as to whether it runs or not.

Reply:
I've updated the second call to send different code (number 1) which is not updated in `req.rstCode`